WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -5,25 +5,54 @@ class JetpackWindowManager: WindowManager {
     /// receives migration flow updates in order to dismiss it when needed.
     private var cancellable: AnyCancellable?
 
+    var shouldImportMigrationData: Bool {
+        return !AccountHelper.isLoggedIn && !UserPersistentStoreFactory.instance().isJPContentImportComplete
+    }
+
     override func showUI(for blog: Blog?) {
         // If the user is logged in and has blogs sync'd to their account
         if AccountHelper.isLoggedIn && AccountHelper.hasBlogs {
-            shouldShowMigrationUI ? showMigrationUI(blog) : showAppUI(for: blog)
+            showAppUI(for: blog)
             return
         }
 
-        // Show the sign in UI if the user isn't logged in
         guard AccountHelper.isLoggedIn else {
-            showSignInUI()
+            if shouldImportMigrationData {
+                importAndShowMigrationContent(blog) { [weak self] in
+                    self?.showSignInUI()
+                }
+            } else {
+                showSignInUI()
+            }
             return
         }
-
         // If the user doesn't have any blogs, but they're still logged in, log them out
         // the `logOutDefaultWordPressComAccount` method will trigger the `showSignInUI` automatically
         AccountHelper.logOutDefaultWordPressComAccount()
     }
 
-    private func showMigrationUI(_ blog: Blog?) {
+    func importAndShowMigrationContent(_ blog: Blog?, failureCompletion: (() -> ())?) {
+        DataMigrator().importData() { [weak self] result in
+            guard let self else {
+                return
+            }
+
+            switch result {
+            case .success:
+                UserPersistentStoreFactory.instance().isJPContentImportComplete = true
+                NotificationCenter.default.post(name: .WPAccountDefaultWordPressComAccountChanged, object: nil)
+                self.showMigrationUIIfNeeded(blog)
+            case .failure:
+                failureCompletion?()
+            }
+        }
+    }
+
+    private func showMigrationUIIfNeeded(_ blog: Blog?) {
+        guard shouldShowMigrationUI else {
+            return
+        }
```

Comment:
I wonder if there's a case where the migration succeeds, but we don't want to show the migration UI? If we're always going to show the migration UI after the import succeeds, I think we can delete this guard check and remove the `shouldShowMigrationUI` variable.

Reply:
@dvdchr The `shouldShowMigrationUI` is also checking whether the feature flag is enabled. If we move that feature flag check to `shouldImportMigrationData`, [which I think we should](https://github.com/wordpress-mobile/WordPress-iOS/pull/19676/files#r1035152245), then we can remove `shouldShowMigrationUI` variable.